src/main/java/mate/academy/internetshop/dao/jdbcimpl/ShoppingCartJdbcImpl.java

```diff
@@ -0,0 +1,150 @@
+import mate.academy.internetshop.exception.DataProcessingException;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.model.ShoppingCart;
+import mate.academy.internetshop.util.ConnectionUtil;
+
+@Dao
+public class ShoppingCartJdbcImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart create(ShoppingCart element) {
+        String sql = "INSERT INTO shopping_cart(user_id) VALUES (?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(sql,
+                    Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, element.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                element.setId(resultSet.getLong(1));
+            }
+            return element;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t add shopping cart to DB", e);
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long element) {
+        String sql = "SELECT * FROM internet-shop.shopping_cart WHERE shopping_cart_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(sql);
+            statement.setLong(1, element);
+            ResultSet resultSet = statement.executeQuery();
+            ShoppingCart shoppingCart = null;
+            while (resultSet.next()) {
+                shoppingCart = getShoppingCartFromResultSet(resultSet);
+            }
+            return Optional.ofNullable(shoppingCart);
+        } catch (SQLException e) {
+            throw new DataProcessingException("can`t receive shopping cart from DB by id ", e);
+        }
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        List<ShoppingCart> shoppingCarts = new ArrayList<>();
+        String sql = "SELECT * FROM shopping_cart;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(sql);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                shoppingCarts.add(getShoppingCartFromResultSet(resultSet));
+            }
+            return shoppingCarts;
+        } catch (SQLException e) {
+            throw new DataProcessingException("can`t receive all shopping cart from DB ", e);
+        }
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart element) {
+        deleteProductFromShoppingCart(element);
+        addProductToShoppingCart(element);
+        return element;
+    }
+
+    @Override
+    public boolean delete(Long element) {
+        String sql = "DELETE FROM internet-shop.shopping_cart WHERE shopping_cart_id=?;";
```

Comment:
```suggestion
        String sql = "DELETE FROM shopping_cart WHERE shopping_cart_id=?;";
```
перед цим треба видаляти зв'язку shopping cart + products?

Reply:
але якщо каскадна модель воно повинно само видалитись?

Reply:
Так